data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/OpenSearchDetectorScanBulkApiWrapperTest.java

```diff
@@ -0,0 +1,124 @@
+            "\"%s\"\n";
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+    @Mock
+    private RestClient restClient;
+    @Mock
+    private Response response;
+    @Mock
+    private HttpEntity httpEntity;
+
+    private OpenSearchDetectorScanBulkApiWrapper wrapper;
+
+    @BeforeEach
+    void setup() {
+        MockitoAnnotations.openMocks(this);
+        when(restHighLevelClient.getLowLevelClient()).thenReturn(restClient);
+        this.wrapper = new OpenSearchDetectorScanBulkApiWrapper(restHighLevelClient);
+    }
+
+    @Test
+    void testBulk() throws IOException {
+        final String index1 = UUID.randomUUID().toString();
+        final String document1 = UUID.randomUUID().toString();
+        final IndexOperation<Object> indexOperation1 = createIndexOperation(index1, document1);
+        final BulkOperation bulkOperation1 = new BulkOperation.Builder()
+                .index(indexOperation1)
+                .build();
+
+        final String index2 = UUID.randomUUID().toString();
+        final String document2 = UUID.randomUUID().toString();
+        final IndexOperation<Object> indexOperation2 = createIndexOperation(index2, document2);
+        final BulkOperation bulkOperation2 = new BulkOperation.Builder()
+                .index(indexOperation2)
+                .build();
+
+        final BulkRequest bulkRequest = new BulkRequest.Builder()
+                .operations(List.of(bulkOperation1, bulkOperation2))
+                .build();
+
+        when(restClient.performRequest(any())).thenReturn(response);
+        when(response.getEntity()).thenReturn(httpEntity);
+        when(httpEntity.getContent()).thenReturn(new ByteArrayInputStream(BULK_RESPONSE_CONTENT.getBytes()));
+
+        final BulkResponse bulkResponse = wrapper.bulk(bulkRequest);
+        assertThat(bulkResponse.errors(), equalTo(false));
+        assertThat(bulkResponse.took(), equalTo(50L));
+        assertThat(bulkResponse.items().size(), equalTo(2));
+        assertThat(bulkResponse.items().get(0).index(), equalTo("test-index-4"));
+        assertThat(bulkResponse.items().get(0).error(), equalTo(null));
+        assertThat(bulkResponse.items().get(0).operationType(), equalTo(OperationType.Index));
+        assertThat(bulkResponse.items().get(0).id(), equalTo("ipUCjI0B4Le7NEG4qhe-"));
+        assertThat(bulkResponse.items().get(0).status(), equalTo(201));
+        assertThat(bulkResponse.items().get(1).index(), equalTo("test-index-4"));
+        assertThat(bulkResponse.items().get(1).error(), equalTo(null));
+        assertThat(bulkResponse.items().get(1).operationType(), equalTo(OperationType.Index));
+        assertThat(bulkResponse.items().get(1).id(), equalTo("i5UCjI0B4Le7NEG4qhe-"));
+        assertThat(bulkResponse.items().get(1).status(), equalTo(201));
+
+        verify(restClient).performRequest(argThat(r -> {
+            assertThat(r.getMethod(), equalTo(DETECTOR_SCAN_API_METHOD));
```

Comment:
There is a lot going on in here, and you have a try-catch block.

I think moving to use an `ArgumentCaptor` would help how much is here. And then you won't need that try-catch either.